doc/wol/Wake-on-LAN-HLD.md

```diff
@@ -88,19 +92,56 @@ Offset   |0     |1     |2     |3     |4     |5     |
          +-----------------------------------------+
 ```
 
+### Magic Pattern in UDP Payload
+**Magic Packet** is a Ethernet frame with structure:
+
+* **IP Header**:
+  * **Destination IP**: user provided target IP address, cloud be IPv4 address or IPv6 address, default is `255.255.255.255`.
+* **UDP Header**:
+  * **Destination Port**: user provided target port, default is 9.
+* **UDP Payload**:
+  * Six bytes of all `0xff`. [6 bytes]
+  * Sixteen repetitions of the target device's MAC address. [96 bytes]
+  * (Optional) A four or six byte password. [4 or 6 bytes]
+
+```
+Byte     |
+Offset   |0     |1     |2     |3     |4     |5     |
+---------+------+------+------+------+------+------+
+           \      \      \      \      \      \     
+         +------+------+------+------+------+------+
+         |.. Destination IP(4 bytes or 16 bytes) ..|
+         +------+------+------+------+------+------+      
+           \      \      \      \      \      \      
+         +------+------+---------------------------+
```

Comment:
Please add a separating line to help user understand which part is IP HEADER and which part is UDP HEADER.

---

```
+------+------+------+------+------+------+      
  \      \      \      \      \      \      
+------+------+---------------------------+   
```
This was used as blank (no byte exist there). If bytes exist in packet but you don't want to highlight them, I suggest to use ellipses (`...`).